src/super_gradients/training/utils/optimizer_utils.py

```diff
@@ -117,6 +126,97 @@ def build_optimizer(net: nn.Module, lr: float, training_params) -> optim.Optimiz
         optimizer_training_params = net_named_params
 
     # CREATE AN OPTIMIZER OBJECT AND INITIALIZE IT
-    optimizer = optimizer_cls(optimizer_training_params, lr=lr, **training_params.optimizer_params)
+    optimizer = optimizer_cls(optimizer_training_params, **training_params.optimizer_params)
+
+    return optimizer
+
+
+def separate_lr_groups(model: nn.Module, lr_dict: Dict[str, float]) -> List[Dict]:
```

Comment:
I just realized this method is missing one important piece of logic which correct handling of weight_decay on bias and normalization layers.

When `training_params.zero_weight_decay_on_bias_and_bn` is enabled, we should create twice the number of parameter groups for each matching prefix. See how it's done in toolbelt https://github.com/BloodAxe/pytorch-toolbelt/blob/develop/pytorch_toolbelt/optimization/functional.py#L69

When we have `lr=dict(default=0.1, head=0.01)`,
we should have `head` and `head_no_wd` parameter groups while the `head_no_wd` should contain bias & normalization params from the head.

Reply:
No it does not miss that logic, the name is pretty explanatory.
We currently handle weight decay through optimizer_params.
The feature here is to assign different lrs. This is more work which also introduces more breaking changes (moving from optmizer_params.weight_decay etc) which I disagree is a part of this feature.

Reply:
It actually does perserve it.
It seperates the groups, but since we unpack **extra_optim_params we just have two groups with the same name.
I actually think its good that way.

Reply:
Alrighty then ^_^